jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -234,35 +232,40 @@ private void authenticatePush() {
   }
 
   private void saveDocker() {
-    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
-        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-
     Optional<DockerClient> dockerClient =
         buildConfiguration.getBaseImageConfiguration().getDockerClient();
     Preconditions.checkArgument(dockerClient.isPresent());
-
-    results.tarPath =
+    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
+        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+    Future<LocalImage> localImageFuture =
+        executorService.submit(
+            new LocalBaseImageSteps(
+                    executorService, buildConfiguration, childProgressDispatcherFactory)
+                .processDockerDaemonBaseImageStep(dockerClient.get()));
```

Comment:
I like the centralization into a single class, but this code looks a bit unwieldy.  Make it a static?

Reply:
Ah, the original reason it was non-static was to avoid copying the `executorService`, `buildConfiguration`, `childProgressDispatcherFactory` parameter list in every method signature. I'll see if I can do something about it though, this pattern is a bit strange.